spring-cloud-generator/scripts/generate-showcase.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+# To VERIFY: ./scripts/generate-showcase.sh
+# To UPDATE: /scripts/generate-showcase.sh -u
+UPDATE=0
+while getopts u flag
+do
+    case "${flag}" in
+        u) UPDATE=1;;
+    esac
+done
+
+# If not set, assume working directory is spring-cloud-generator
+if [[ -z "$SPRING_GENERATOR_DIR" ]]; then
+  SPRING_GENERATOR_DIR=`pwd`
+fi
+SPRING_ROOT_DIR=${SPRING_GENERATOR_DIR}/..
+SHOWCASE_STARTER_OLD_DIR=${SPRING_GENERATOR_DIR}/showcase/showcase-spring-starter
+SHOWCASE_STARTER_NEW_DIR=${SPRING_GENERATOR_DIR}/showcase/showcase-spring-starter-generated
+
+# $1 - directory containing existing showcase-spring-starter (golden)
+# $2 - directory containing newly generated showcase-spring-starter
+function verify(){
+  OLD_DIR=$1
+  NEW_DIR=$2
+  SHOWCASE_STARTER_DIFF=$(diff -r ${NEW_DIR}/src/main ${OLD_DIR}/src/main)
+  SHOWCASE_STARTER_POM_DIFF=$(diff -r ${NEW_DIR}/pom.xml ${OLD_DIR}/pom.xml)
+  if [ "$SHOWCASE_STARTER_DIFF" != "" ] || [ "$SHOWCASE_STARTER_POM_DIFF" != "" ]
+  then
+      echo "Differences detected in generated showcase starter module: "
+      echo "Diff from src/main: "
+      echo $SHOWCASE_STARTER_DIFF
+      echo "Diff from pom.xml: "
+      echo $SHOWCASE_STARTER_POM_DIFF
+      exit 1;
+  else
+      echo "No differences found in showcase-spring-starter"
+      rm -r ${NEW_DIR}
+  fi
+}
+
+# $1 - target directory for generated starter
+function generate_showcase_spring_starter(){
+  # do something
+  SHOWCASE_STARTER_DIR=$1
+
+  # Compute the parent project version.
+  cd ${SPRING_ROOT_DIR}
+  PROJECT_VERSION=$(./mvnw help:evaluate -Dexpression=project.version -q -DforceStdout)
+  cd ${SPRING_GENERATOR_DIR}
+  GAPIC_GENERATOR_JAVA_VERSION=$(./../mvnw help:evaluate -Dexpression=gapic-generator-java-bom.version -q -DforceStdout)
+
+  # Install local snapshot jar for spring generator
+  cd ${SPRING_GENERATOR_DIR} && mvn install
+
+  # Clone sdk-platform-java (with showcase library)
+  git clone https://github.com/googleapis/sdk-platform-java.git
+  git checkout "v${GAPIC_GENERATOR_JAVA_VERSION}"
+
+  # Alternative considered: if showcase client library is available on Maven Central,
+  # Instead of downloading sdk-platform-java/showcase (for client library, and generation setup),
+  # Can instead download googleapis (for generation setup) and gapic-showcase (for protos)
+
+  # Install showcase client libraries locally
+  cd sdk-platform-java && mvn clean install -B -ntp -DskipTests -Dclirr.skip -Dcheckstyle.skip
+  cd showcase && mvn clean install
+
+  # Modify sdk-platform-java/WORKSPACE adapted from setup-googleapi-rules.sh
```

Comment:
Note on enhancement: the working generation/post-processing steps here are adapted from generator scripts for actual modules; can try to parameterize and reuse more after https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1956 moves them into functions